ciao-controller/client.go

```diff
@@ -114,26 +109,84 @@ func (client *ssntpClient) releaseResources(instanceID string) error {
 	return nil
 }
 
-func (client *ssntpClient) removeInstance(instanceID string) {
-	err := client.releaseResources(instanceID)
+func (client *ssntpClient) removeInstance(i *types.Instance) {
+	err := client.releaseResources(i)
 	if err != nil {
 		glog.Warningf("Error when releasing resources for deleted instance: %v", err)
 	}
-	client.deleteEphemeralStorage(instanceID)
-	err = client.ctl.ds.DeleteInstance(instanceID)
+	client.deleteEphemeralStorage(i.ID)
+	err = client.ctl.ds.DeleteInstance(i.ID)
 	if err != nil {
 		glog.Warningf("Error deleting instance from datastore: %v", err)
 	}
+
+	if i.CNCI {
+		return
+	}
+
+	// check the cnci after 5 minutes has elapsed to see if it's idle.
+	go func() {
```

Comment:
If I do a ciao-cli instance delete -all on a 10000 node cluster will I end up with 10000 of these go routines?  This might not be a critical problem if checkCNCI is re-entrant but you'll probably end up with loads of warnings in the logs.